### DISCUSSION TITLE
Sharing Public Tasks

### DISCUSSION THREAD

**Discussion by aljawaid** [kanboard/kanboard#5354] [Q&A] [answered]
Hello everyone and @fguillot 

Based on https://github.com/kanboard/kanboard/issues/2629#issuecomment-336232773 and #5344 and https://kanboard.discourse.group/t/how-to-suppress-back-to-pinboard-in-public-view/2786/11, can somebody clarify what needs to be done?

### Currently
The token is saved in the projects table and pulled through a model for that. So the token is unique per project, meaning any task in that project shares the token.

### Proposed for Single Tasks
I dont know if I am correct but I wanted some advice if this will work.

To make the task public and ONLY the task, with no 'back to kanboard' links etc.
 - create a new token in the tasks table for each task OR create a new table containing task id and a new token.
   - db table
 - generate token for each existing tasks in the table
   - model (not sure which functions to copy or just `readonly`)
 - add new link into the template to allow the option of sharing the task on its own
   - model and template... maybe controller
  
If I can understand the steps, maybe slowly I can get it into a plugin as the feature is highly needed for some people including myself.

**Comment by fguillot** [comment, marked as answer]
Here is what I would do:

- Add a new column in the `tasks` table (might be easier), but having a new table should work as well. 
- Add a new button in the task view to allow people sharing and stop sharing individual tasks. This button will likely trigger actions related to sharing and generating tokens.
- Generate a new unique token only when a user decides to share an individual task. There's no need to generate tokens for all tasks in the project, as this could be resource-intensive and unnecessary.
- Modify the method `TaskViewController::readonly()` to accept an optional task token. This ensures that existing URLs continue to work without any issues while allowing for task-specific sharing.
- Do not show the link "Back to the board" if there is no project token.

> I can get it into a plugin as the feature is highly needed for some people including myself

If you are willing to develop such feature, I encourage you to contribute directly to Kanboard instead of developing a new plugin because it would benefit to more people.

**Reply by aljawaid** [reply]
I have an incomplete plugin that this could fit into but thats a lot of unknown steps for me the way you have described it with all the options.

I will have a deep look into it though; never say never.

**Comment by alfredbuehler** [comment]
IIRC, another issue is (or was?), if a shared task contains attachments, pics, or other assets. Would this be addressed also by the proposed solution?
